Scripts/OpenMania/Stage/Common/Ring.hsl

```diff
@@ -218,21 +218,21 @@ class Ring {
         var sensor, halfW = this.HitboxHalfW, halfH = this.HitboxHalfH;
         // Moving left
         if (this.XSpeed < 0.0 && ( sensor = TileCollision.Line(this.X, this.Y, 3, halfW, this.PlaneIndex, -1) )) {
-            if (sensor["Collided"]) {
+            if (this.SensorCollided) {
                 this.X = sensor["X"] + halfW;
```

Comment:
Might wanna fix the `sensor["X"]` and `sensor["Y"]` stuff on this line 222, 229, 236, 245, and wherever else you can Find `sensor["` just in case